test/components/project/list.spec.js

```diff
@@ -327,17 +359,168 @@ describe('ProjectList', () => {
+        [],
+        [
+          [{}, {}, {}, {}, {}],
+          [{}, {}, {}, {}, {}, {}],
+          [{}, {}, {}, {}, {}]
+        ]
+      );
+      const component = mountComponent();
+      component.findAllComponents(DatasetRow).length.should.equal(15);
+      const blocks = component.findAllComponents(ProjectHomeBlock);
+      blocks[0].props().maxDatasets.should.equal(5);
+      blocks[0].findAllComponents(DatasetRow).length.should.equal(5);
+      blocks[1].findAllComponents(DatasetRow).length.should.equal(5);
+      blocks[2].findAllComponents(DatasetRow).length.should.equal(5);
+    });
+
+    it('uses the same datasets limit across home blocks', () => {
+      createProjects(
+        [
+          { name: 'Project 1' },
+          { name: 'Project 2' },
+          { name: 'Project 3' },
+          { name: 'Project 4' }
+        ],
+        [],
+        [
+          [{}, {}, {}, {}],
+          [{}, {}, {}, {}],
+          [{}, {}, {}, {}],
+          [{}, {}, {}, {}]
+        ]
+      );
+      const component = mountComponent();
+      component.findAllComponents(DatasetRow).length.should.equal(12);
+      const blocks = component.findAllComponents(ProjectHomeBlock);
+      blocks[0].props().maxDatasets.should.equal(3);
+      blocks[0].findAllComponents(DatasetRow).length.should.equal(3);
+      blocks[1].findAllComponents(DatasetRow).length.should.equal(3);
+      blocks[2].findAllComponents(DatasetRow).length.should.equal(3);
+      blocks[3].findAllComponents(DatasetRow).length.should.equal(3);
+    });
+
+    it('renders correctly with a small number of datasets where one project is above the original limit', () => {
+      createProjects(
+        [
+          { name: 'Project 1' },
+          { name: 'Project 2' }
+        ],
+        [],
+        [
+          [{}, {}, {}, {}],
+          [{}, {}]
+        ]
+      );
+      const component = mountComponent();
+      component.findAllComponents(DatasetRow).length.should.equal(6);
+      const blocks = component.findAllComponents(ProjectHomeBlock);
+      blocks[0].props().maxDatasets.should.equal(4);
+      blocks[0].findAllComponents(DatasetRow).length.should.equal(4);
+      blocks[1].findAllComponents(DatasetRow).length.should.equal(2);
+    });
+
+    it('it renders correctly with a small number of datasets', () => {
+      createProjects(
+        [
+          { name: 'Project 1' },
+          { name: 'Project 2' }
+        ],
+        [],
+        [
+          [{}, {}, {}],
+          [{}, {}]
+        ]
+      );
+      const component = mountComponent();
+      component.findAllComponents(DatasetRow).length.should.equal(5);
+      const blocks = component.findAllComponents(ProjectHomeBlock);
+      blocks[0].props().maxDatasets.should.equal(3);
+      blocks[0].findAllComponents(DatasetRow).length.should.equal(3);
+      blocks[1].findAllComponents(DatasetRow).length.should.equal(2);
+    });
+  });
+
```

Comment:
Tests are needed for the combined form + dataset case. 

- over the limit: more than 15 forms + datasets get cut off
- under the limit: all forms and datasets shown
- a mix: more forms than datasets or something so that forms get cutoff but datasets don't, or something like that